docs/git.rst

```diff
@@ -0,0 +1,611 @@
+.. -*- coding: utf-8 -*-
+
+==================
+git best practices
+==================
+Collection of best practices that would help the overall community to work with git more efficiently.
+
+Basics
+======
+Some introductory definitions and concepts,
+if you are already familiar enough with git,
+head to next section: :ref:`general-guidelines-label`.
+
+Working model
+-------------
+In ``git``
+(as well as all modern `DVCS <http://en.wikipedia.org/wiki/Distributed_revision_control>`_),
+distributing changes to others is a two steps process
+(contrary to traditional VCS like ``svn``).
+
+This way what on svn is a single ``svn ci`` in git is two commands:
+``git commit`` and ``git push``.
+
+This may seem to be a drawback,
+but instead it's a feature.
+
+**You are working locally until you decide to push your changes.**
+
+Not a single commit anymore,
+but a series of them,
+meaning that all those fears,
+concerns,
+doubts are taken away!
+
+You can freely fix/change/remove/rework/update/... your commits afterwards.
+
+Just push your changes whenever you are sure they are what you,
+and others,
+expect them to be.
+
+Concepts
+--------
+In git there are:
+
+commits
+   A patch made out of changes (additions, removals) on files tracked by git.
+
+branches
+   Series of commits that have a name.
+
+tags
+   A name attached to a single commit.
+
+``HEAD``
+   A pointer that always tells you where you are
+   (extremely useful when doing some operations).
+
+The index
+  A temporal staging storage with changes on files that are pending to be added to a commit.
+  If your git output is colored,
+  green filenames are those in the index.
+
+Working tree
+  Your current modified files.
+  This is the only place where you can loose your changes.
+  If your git output is colored,
+  red filenames are those in the working tree.
+
+Stash
+  Temporal storage for changes,
+  again,
+  extremely useful in some scenarios,
+  see further below for examples.
+
+Branches
+--------
+Another great feature of DVCS is cheap branching,
+i.e. make it effortlessly and really useful to use branches.
+No more working on the master branch always as is too much effort to work on branches.
```

Comment:
Possibly consider changing to:
"As it's no longer too much effort to branch, there is no need to always work on the master branch. A developer can branch easily for each fix/feature."